File: clients/js/examples/01_hello.js
// 01_hello.js — connect, ping, insert, find — smallest possible Node.js example.
const { OxiDB } = require("../index.js");

(async () => {
  const db = new OxiDB("http://127.0.0.1:9080");

  console.log("ping:", await db.ping());

  await db.collection("users").insert({
    name: "Alice", age: 30, email: "alice@example.com",
  });

  const alice = await db.collection("users").findOne({ name: "Alice" });
  console.log("loaded:", alice);
})().catch((e) => { console.error(e); process.exit(1); });

File: clients/julia/examples/01_hello.jl
#!/usr/bin/env julia
# 01_hello.jl — connect, insert, read back, close.
# Smallest possible Julia program against OxiDB.

using OxiDb

OxiDb.connect("127.0.0.1", 4444) do db
    @show ping(db)

    insert(db, "users", Dict(
        "name"  => "Alice",
        "age"   => 30,
        "email" => "alice@example.com",
    ))

    alice = find_one(db, "users", Dict("name" => "Alice"))
    println("Loaded: ", alice)
end
